internal/server/server.go

```diff
@@ -80,6 +79,7 @@ type ListenerOptions struct {
 type UIOptions struct {
 	Enabled  bool
 	ProxyURL types.URL `mapstructure:"proxyURL"`
+	FS       fs.FS     `mapstructure:"-"`
```

Comment:
This shouldn't really need to exist but was added for injecting a fake `fs.FS` in tests – open to a better alternative if one exists!

Reply:
A shim like this is a good option, but I think in this case we might only be testing this shim.

I wonder if we can use `uiFS` and patch that to a different value in tests, to bring the tests closer to production.

Reply:
What's the behaviour if the requested file isn't found, i.e. if someone neglects to generate the UI for dev?

Reply:
It returns a 404 with page body `404 not found`

Reply:
Using a package-level variable as the shim didn't work because the variable needs to be an `embed.FS` for `go:embed` to work, and we can't create an `embed.FS` with actual files in it. 

So instead I changed everything to read `options.UI.FS` and ignore the package level variable.